squirrel.frontier/src/main/java/org/dice_research/squirrel/data/uri/filter/MongoDBKnowUriFilter.java

```diff
@@ -41,188 +42,208 @@
-    public void addHashValuesForUris(List<CrawleableUri> uris) {
-
-    }
-    
-    
-    public void purge() {
-    	mongoDB.getCollection(COLLECTION_NAME).drop();
-    }
-
-    @Override
-    public List<CrawleableUri> getOutdatedUris() {
-    	// get all uris with the following property:
-        // (nextCrawlTimestamp has passed) AND (crawlingInProcess==false OR lastCrawlTimestamp is 3 times older than generalRecrawlTime)
-    	
-    	long generalRecrawlTime = Math.max(FrontierImpl.DEFAULT_GENERAL_RECRAWL_TIME, FrontierImpl.getGeneralRecrawlTime());
-
-    	Bson filter = Filters.and(Filters.eq("COLUMN_TIMESTAMP_NEXT_CRAWL", System.currentTimeMillis()),
-    			Filters.or(
-	    			Filters.eq("COLUMN_CRAWLING_IN_PROCESS", false),
-	    			Filters.eq("COLUMN_TIMESTAMP_LAST_CRAWL", System.currentTimeMillis() - generalRecrawlTime * 3)
-    			));
-
-        Iterator<Document> uriDocs = mongoDB.getCollection(COLLECTION_NAME).find(filter).iterator();
-
-        List<CrawleableUri> urisToRecrawl = new ArrayList<>();
-        while (uriDocs.hasNext()) {
-            try {
-                Document doc = uriDocs.next();
-                String ipString = (String) doc.get(COLUMN_IP);
-                if (ipString.contains("/")) {
-                    ipString = ipString.split("/")[1];
-                }
-                urisToRecrawl.add(new CrawleableUri(new URI((String) doc.get(COLUMN_URI)), InetAddress.getByName(ipString)));
-            } catch (URISyntaxException | UnknownHostException e) {
-                LOGGER.warn(e.toString());
-            }
-        }
-
-        // mark that the uris are in process now
-        for (CrawleableUri uri : urisToRecrawl) {
-        	
-        	BasicDBObject newDocument = new BasicDBObject();
-        	newDocument.append("$set", new BasicDBObject().append(COLUMN_CRAWLING_IN_PROCESS, true));
-        	
-        	BasicDBObject searchQuery = new BasicDBObject().append(COLUMN_URI,  uri.getUri().toString());
-        	
-        	 mongoDB.getCollection(COLLECTION_NAME).updateMany(searchQuery, newDocument);
-        	
-        }
+	}
+
+	public MongoDBKnowUriFilter(String hostName, Integer port, int max_depth) {
+		this(hostName, port);
+		this.max_depth = max_depth;
+	}
+
+	@Override
+	public boolean isUriGood(CrawleableUri uri) {
+		MongoCursor<Document> cursor = mongoDB.getCollection(COLLECTION_NAME)
+				.find(new Document("uri", uri.getUri().toString())).iterator();
+
+		if (cursor.hasNext()) {
+			LOGGER.debug("URI {} is not good", uri.toString());
+			Document doc = cursor.next();
+			Long timestampRetrieved = Long.parseLong(doc.get(COLUMN_TIMESTAMP_LAST_CRAWL).toString());
+			cursor.close();
+			if ((System.currentTimeMillis() - timestampRetrieved) < recrawlEveryWeek) {
+				return false;
+			} else {
+				return true;
+			}
+		} else {
+
+			if (uri.getData().containsKey(Constants.URI_DEPTH)) {
+				int depth = Integer.parseInt(uri.getData(Constants.URI_DEPTH).toString());
+				if (depth > max_depth) {
```

Comment:
Only if the depth is enabled and if it is present on URI data, which means that the user wants to filter by depth also.  Actually,i thought about something like you suggested, but i do not see the depth as an entire new filter, since the KnowUriFilter prevents the same URI from being
crawled twice in the short time. I think that the depth as an option in the KnowUriFilter makes more sense. Actually, it makes more sense as an option for all kind of filters, thats why i was thinking about including methods on the Interface or even create an abstract class to force all the filters to have it.

Reply:
> Actually, it makes more sense as an option for all kind of filters

But that is exactly the issue. You start to create filters with combinations of constraints they use to filter. Instead, the filters should focus on exactly 1 condition and the user should be allowed to combine these very simple filters.